### PR TITLE
#974 a generic cache implementation

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #974)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -42,6 +42,8 @@ versions["commonNet"] = project.hasProperty('commonNet.version') ? project.prope
 versions["sshd"] = project.hasProperty('sshd.version') ? project.properties['sshd.version'] : "2.2.0"
 // commonIo has included by apache hadoop so we use the identical version
 versions["commonsIo"] = project.hasProperty('commonsIo.version') ? project.properties['commonsIo.version'] : "2.4"
+// commonIo has included by apache kafka so we use the identical version
```
Comment (saivirtue) [inline gradle/dependencies.gradle]: typo? should we use guava here?

File: ohara-common/src/main/java/com/island/ohara/common/cache/RefreshableCache.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    private Cache<K, V> cache = null;
+    private Supplier<Map<K, V>> supplier = null;
+    private Duration frequency = Duration.ofSeconds(5);
+
+    private Builder() {}
+
+    public Builder<K, V> cache(Cache<K, V> cache) {
+      this.cache = Objects.requireNonNull(cache);
+      return this;
+    }
```
Comment (saivirtue) [inline ohara-common/src/main/java/com/island/ohara/common/cache/RefreshableCache.java]: 看起來`RefreshableCache`是extend `Cache`的interface，在建立constructor時還需要傳入Cache物件嗎?

Comment (chia7712) [reply ohara-common/src/main/java/com/island/ohara/common/cache/RefreshableCache.java]: yep, RefreshableCache compose of cache and a thread which keeps updating cache. Accepting a cache object can reduce the duplicate code. It means we don't need to accept the options same to Cache.Builder.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (chia7712) [reply gradle/dependencies.gradle]: YOU ARE RIGHT :)

**Pull request merged**